[페이지] TrainerList 페이지 추가

Ho2eny opened the pull request:
### 요구 사항
| Design | Develop 1 | Develop 2|
|-|-|-|
|![image](https://user-images.githubusercontent.com/50590025/189029539-f285910a-cd7f-422f-8889-8fece2940e34.png)|![image](https://user-images.githubusercontent.com/50590025/189029774-67e5f40d-df58-444a-9cfb-c6c08e6e13f6.png)|![image](https://user-images.githubusercontent.com/50590025/189029801-46057ffa-e4e9-4183-8fd2-25574b776745.png)

### 수정 내역
- TrainerList 페이지를 추가합니다
- (**22-10-02**) 변경된 `MainHeader`를 적용합니다. 빌드 에러를 막기 위함입니다
![image](https://user-images.githubusercontent.com/50590025/193439722-e9eb57d1-2e89-4cd5-af47-b0da9cdcf75b.png)
- (**22-10-02**) `TrainerItem`이 BottomNavigation에 가리지 않기 위해 `margin-bottom:66px`을 추가합니다
![image](https://user-images.githubusercontent.com/50590025/193439757-c15e985d-930a-438c-8726-e50a37ed1a5e.png)


### 당부의 말씀
- ~~BottomNavigation을 제외하고 영역을 잡고 싶은데, 그 부분을 잘 처리 하지 못해서 사진의 가장 오른쪽 마지막 TrainerItem이 Navigation 뒤에 숨어지게 되는 문제가 있습니다.  어떻게 해결하는 것이 좋을까요, 도움이 필요합니다~~

Ho2eny committed:
feat: clssName Props 추가, Props export 추가

className Props 추가
Props export 추가

Ho2eny committed:
feat: TrainerListPage 추가

TrainerListPage 추가

devhwi commented:
바텀 네비만큼의 공갈 div를 주는건 너무 하드코딩일까요?

아니면 리스트들을 감싸고 있는 영역에서 마진이나 패딩 추가..??

Ho2eny commented:
> 바텀 네비만큼의 공갈 div를 주는건 너무 하드코딩일까요?
> 
> 아니면 리스트들을 감싸고 있는 영역에서 마진이나 패딩 추가..??

바닥에 Navigation height만큼 높이를 줬는데, 위로 안올라오더라구요,, 제가 잘 못해서인징

wlsdhr0831 commented:
> > 바텀 네비만큼의 공갈 div를 주는건 너무 하드코딩일까요?
> > 아니면 리스트들을 감싸고 있는 영역에서 마진이나 패딩 추가..??
> 
> 바닥에 Navigation height만큼 높이를 줬는데, 위로 안올라오더라구요,, 제가 잘 못해서인징

회사에서도 그냥 네비 높이만큼 바닥에 padding 값 주는 것 같아염

Ho2eny committed:
feat: MainHeader를 업데이트합니다.

- AppLog 이미지를 추가합니다.
- UserProfileImage 를 추가합니다

Ho2eny committed:
feat: 변경된 MainHeader 적용

변경된 MainHeader를 적용합니다

Ho2eny committed:
fix: TrainerItemContainer 추가

- BottomNavigation에 TrainerItem 내용이 가리는 현상 수정
- Navigation Height, TrainerItem 사이 간격이 변경될 때마다 수정이 필요함

Ho2eny commented:
업데이트된 [`MainHeader`](https://github.com/TheNewestWorld/groove-front/pull/55)를 적용했어요, 머지된 후에 기존의 `MainHeader`에서 빌드 에러가 발생하지 않도록 하기 위함입니다

Ho2eny commented:
`BottomNavigation`으로 부터 `margin`을 넣어서 위로 올렸습니다. [styles](https://github.com/TheNewestWorld/groove-front/pull/47/files#diff-08b08b500c6656e740c3dc98e31ce5868bdac08df3f5400823184d568d657d25R7-R15)
네비게이션 높이, 아이템 간의 간격을 가져와서 계산하면 좋을텐데, 네비게이션 높이나 아이템 사이 간격이 변경되면 이 부분에 수정을 요구할 것 같습니다

wlsdhr0831 commented:
> `BottomNavigation`으로 부터 `margin`을 넣어서 위로 올렸습니다. [styles](https://github.com/TheNewestWorld/groove-front/pull/47/files#diff-08b08b500c6656e740c3dc98e31ce5868bdac08df3f5400823184d568d657d25R7-R15)
> 
> 네비게이션 높이, 아이템 간의 간격을 가져와서 계산하면 좋을텐데, 네비게이션 높이나 아이템 사이 간격이 변경되면 이 부분에 수정을 요구할 것 같습니다
> 
> 

근데 네비게이션 없어지는거 아닌가요???

Ho2eny commented:
> 
네, 네비게이션은 없어지는데,
없어진 `Trainer`  쪽 PR이고 수정만해서 머지 시켜놓는 거라 네비게이션을 빼지는 않았습니다.

wlsdhr0831 reviewed (approved):
👍 👍 
우선 순위가 낮아져 진행하지 않고 일단 작업 마무리용으로
최신 수정사항 반영하지 않음 확인하였습니다~
